src/clients/c/tb_client/context.zig

```diff
@@ -134,15 +134,22 @@ pub fn ContextType(
             assert(context.client_id != 0); // Broken CSPRNG is the likeliest explanation for zero.
 
             log.debug("{}: init: parsing vsr addresses: {s}", .{ context.client_id, addresses });
-            context.addresses = vsr.parse_addresses(
-                context.allocator,
+            context.addresses = .{};
+            const addresses_parsed = vsr.parse_addresses(
                 addresses,
-                constants.replicas_max,
+                context.addresses.unused_capacity_slice(),
             ) catch |err| return switch (err) {
                 error.AddressLimitExceeded => error.AddressLimitExceeded,
-                else => error.AddressInvalid,
+                error.AddressHasMoreThanOneColon,
+                error.AddressHasTrailingComma,
+                error.AddressInvalid,
+                error.PortInvalid,
+                error.PortOverflow,
+                => error.AddressInvalid,
```

Comment:
We were eating up an allocation failure here before, so I figured I'd list them all explicitly!